packages/protocol/tasks/utils.ts

```diff
@@ -84,6 +86,67 @@ const MessageStatus = {
     FAILED: 3,
 }
 
+async function getLatestBlockHeader(hre: any) {
+    const block: Block = await hre.ethers.provider.send(
+        "eth_getBlockByNumber",
+        ["latest", false]
+    )
+
+    const logsBloom = block.logsBloom.toString().substring(2)
+
+    const blockHeader: BlockHeader = {
+        parentHash: block.parentHash,
+        ommersHash: block.sha3Uncles,
+        beneficiary: block.miner,
+        stateRoot: block.stateRoot,
+        transactionsRoot: block.transactionsRoot,
+        receiptsRoot: block.receiptsRoot,
+        logsBloom: logsBloom.match(/.{1,64}/g)!.map((s: string) => "0x" + s),
+        difficulty: block.difficulty,
+        height: block.number,
+        gasLimit: block.gasLimit,
+        gasUsed: block.gasUsed,
+        timestamp: block.timestamp,
+        extraData: block.extraData,
+        mixHash: block.mixHash,
+        nonce: block.nonce,
+        baseFeePerGas: block.baseFeePerGas ? parseInt(block.baseFeePerGas) : 0,
+    }
+
+    return { block, blockHeader }
+}
+
+async function getSignalProof(
+    hre: any,
+    contractAddress: string,
+    key: string,
+    blockHash: string,
+    blockHeader: BlockHeader
+) {
+    const proof: EthGetProofResponse = await hre.ethers.provider.send(
+        "eth_getProof",
+        [contractAddress, [key], blockHash]
```

Comment:
Seems the latest parameter should be an integer block number? 
Ref: https://eips.ethereum.org/EIPS/eip-1186

Reply:
Just switched it to block.number in the most recent push, please have a look!